### PR TITLE
ENH: make MRC reader more robust wrt broken axis mapping

### PR DESCRIPTION
**Pull request by kohr-h**
Simple fix for a case when MRC files don't properly define the axis mapping fields.

### CODE REVIEW

File: odl/tomo/data/mrc.py [hunk 1]
```diff
@@ -346,7 +346,14 @@ def data_axis_order(self):
         except KeyError:
             return (0, 1, 2)
         else:
-            return tuple(int(m) - 1 for m in [mapc, mapr, maps])
+            axis_order = tuple(int(m) - 1 for m in [mapc, mapr, maps])
+            if (len(set(axis_order)) != 3 or
+                    any(m not in (0, 1, 2) for m in axis_order)):
+                # Ignore invalid entries in the header, e.g. 0, 0, 0.
```
Comment (adler-j) [inline odl/tomo/data/mrc.py]: I am strongly against silently improving data. Add something like

```
warnings.warn('Invalid axis_order, using (0, 1, 2)')
```

File: odl/tomo/data/mrc.py [hunk 2]
```diff
@@ -346,7 +346,14 @@ def data_axis_order(self):
         except KeyError:
             return (0, 1, 2)
         else:
-            return tuple(int(m) - 1 for m in [mapc, mapr, maps])
+            axis_order = tuple(int(m) - 1 for m in [mapc, mapr, maps])
+            if (len(set(axis_order)) != 3 or
```
Comment (adler-j) [inline odl/tomo/data/mrc.py]: `if sorted(axis_order) != (0, 1, 2):` 

would perhaps be easier to read?

File: odl/tomo/data/mrc.py [hunk 3]
```diff
@@ -346,7 +346,14 @@ def data_axis_order(self):
...
+                    any(m not in (0, 1, 2) for m in axis_order)):
+                # Ignore invalid entries in the header, e.g. 0, 0, 0.
+                # Some MRC files out there are like that.
+                return (0, 1, 2)
```
Comment (adler-j) [inline odl/tomo/data/mrc.py]: I would prefer going from here as

```
    axis_order = (0, 1, 2)

return axis_order
```

more readable with single return.

File: odl/tomo/data/mrc.py (hunk 1 above)
Comment (kohr-h) [inline odl/tomo/data/mrc.py]: Fair enough

**Comment by kohr-h** [comment]
Fixed, merging after CI.

**Pull request merged**